packages/synapse-react-client/src/components/JsonSchemaForm/templates/BaseInputTemplate.tsx

```diff
@@ -81,12 +82,17 @@ export default function BaseInputTemplate<
       JSON.stringify(schema.default || schema.const) + ' (derived)'
   }
 
+  const placeholder = (registry.formContext as CustomFormContext)
+    .showDerivedAnnotationPlaceholder
+    ? derivedValuePlaceholder || placeholderFromProps
+    : placeholderFromProps
```

Comment:
allow opt-out of derived annotations UI feature

Reply:
Consider breaking up into a few lines to reduce cognitive complexity (the tertiary plus a fallback value)